### ISSUE TITLE
Reinstalling is not working

### ISSUE THREAD

**Issue by hardlydifficult** [hardlydifficult/Miner#20]
Several times we have been unsuccessful updating to a new version.  No idea where it is failing.  The install seems to run okay, but the app crashes on launch.

It's not just Auto-Update, manual re-install does not work either...

Fixed by uninstalling and deleting the app directory.  Then install again.

**Comment by hardlydifficult** [comment]
on it

**Comment by hardlydifficult** [comment]
I added an install.log file which is created when the auto-update is run.  With this, we can hopefully get some data about this issue if this happens again.

Closing until we have a repro again.

**Comment by hardlydifficult** [comment]
This does not appear to be an installation issue... but we need one more repro to be sure.

e.g. possible scenario... the miner is still running while the installer starts copying files.  One of those copies fails because the file is in use.  Then the DLL version mismatch prevents a launch.

Once the issue starts, delete everything is the only fix I know of so far.

Re-open - next step is to test an update again with something that changed that could cause a fail...

**Comment by mongomannen** [comment]
Was able to reproduce. I am 100% sure the miner was not running when the installer started.

Made a diff between the failed* upgrade and a clean install, the following files were not updated by the installer:
AutoUpdater.NET.dll
HardlyCommon.dll
HardlyCommonWindows.dll
Miner.App.dll
Miner.Common.dll
Miner.Middleware.dll
Miner.Middleare.Xmr-stak-cpu.exe
Miner.OS.dll
Miner.OS.Windows.dll

*) No installer error but miner.exe crashes on launch.

I'm just guessing here, but it might be that MSI skips copying the files because the file versions haven't changed.

**Comment by mongomannen** [comment]
Confirming that the file versions are the issue. Ran the installer with logging enabled, `msiexec /l*vx log.txt` and it contains several lines like
`MSI (s) (24:48) [13:40:57:433]: File: C:\Users\mike\AppData\Local\Hardly Difficult LLC\Hardly Difficult Miner\HardlyCommon.dll;	Won't Overwrite;	Won't patch;	Existing file is of an equal version
`

I've checked both .msi files and the two HardlyCommon.dll are different but the file version is the same. Dependencies need to have their version bumped every time they change.
